test/ranges_test.rb

```diff
@@ -0,0 +1,40 @@
+require_relative 'test_helper'
+
+describe "Date Range" do
+  before do
+    @range = Hotel::DateRange.new(start_date: "3 Mar", end_date: "5 Mar")
+    @range2 = Hotel::DateRange.new(start_date: "4 Mar", end_date: "6 Mar")
+  end
+
+  describe "#initialize" do
+    it "creates an instance of DateRange class" do
+      expect(@range).must_be_instance_of Hotel::DateRange
+    end
+
+    it "raises an Argument Error if the end date is less than or equal to start date" do
+      expect{Hotel::DateRange.new(start_date: "7 Mar 2020", end_date: "5 Mar 2020")}.must_raise ArgumentError
+    end
+  end
+
+  describe "#overlap?" do
```

Comment:
There are many different ways two date ranges can overlap. You should make sure to test for all these different ways. These are the tests from the `design-scaffold` branch.

```
  describe "overlap?" do
    before do
      start_date = Date.new(2017, 01, 01)
      end_date = start_date + 3

      @range = Hotel::DateRange.new(start_date, end_date)
    end

    it "returns true for the same range" do
      start_date = @range.start_date
      end_date = @range.end_date
      test_range = Hotel::DateRange.new(start_date, end_date)

      expect(@range.overlap?(test_range)).must_equal true
    end

    xit "returns true for a contained range" do
    end

    xit "returns true for a range that overlaps in front" do
    end

    xit "returns true for a range that overlaps in the back" do
    end

    xit "returns true for a containing range" do
    end

    xit "returns false for a range starting on the end_date date" do
    end

    xit "returns false for a range ending on the start_date date" do
    end

    xit "returns false for a range completely before" do
    end

    xit "returns false for a date completely after" do
    end
  end

```